src/types.ts

```diff
@@ -1198,33 +1198,46 @@ export class IntersectionType<T extends AnyType, K extends AnyType> extends Type
 }
 
 type ValueOf<T> = T[keyof T];
+type EnumCoerceOptions = 'upper' | 'lower';
+export type EnumOptions<T> = {
```

Comment:
could you add `coerce` and `default` methods to EnumType? I like enabling fluent syntax in the API